logline.go

```diff
@@ -205,3 +205,38 @@ func (l *LogLine) Meta(key string) (interface{}, bool) {
 func (l *LogLine) Cache() *Cache {
 	return &l.cache
 }
+
+// Copy creates and returns a copy of the current log line.
+func (l *LogLine) Copy() Record {
+	// Copy metadata
+	md := make(Metadata)
+	for k, v := range l.meta {
+		md[k] = v
+	}
+
+	cpy := &LogLine{
+		cache:          l.cache,
+		wcnt:           l.wcnt,
+		meta:           md,
+		FieldSeparator: l.FieldSeparator,
+	}
+
+	if cpy.wcnt != 0 {
+		// If the log line has been modified, benchmarks have proven
+		// that it's more efficient to serialize and reparse to perform
+		// a copy (both in terms of time and allocation.
+		cpylen := len(l.data) + len(l.data)/2
```

Comment:
please add a brief comment to explain why you choose to use this length

Reply:
and, as you said, the `ToText` fn already allocates the right amount of memory when it calculates `wlen` so you can probably avoid using `cpylen`

Reply:
Yes, It's probably safe to keep the same length as org.data and let ToText takes care of the actual length

Reply:
ToText only estimates the actual length in order to reduce the number of allocations, here is the results of various benchmarks looking at various size for the pre-allocated buffer, for example `bench.len+50` means the benchmark where the preallocated buffer is the original + length + 50% (to account for written fields).
bench.len+50 (current code) is the best all-round (cpu, allocations bytes + count)

```
name \ time/op        bench.len+50  bench.len+25  bench.len+10  bench.len    bench.nil
LogLineCopy/set=0-8    4.73µs ± 5%   5.03µs ± 2%   4.62µs ± 8%  4.74µs ± 2%  4.67µs ± 3%
LogLineCopy/set=1-8    30.4µs ± 8%   31.7µs ± 6%   31.2µs ± 2%  32.6µs ± 5%  31.1µs ± 0%
LogLineCopy/set=10-8   30.3µs ± 4%   31.0µs ± 4%   32.6µs ± 1%  32.7µs ± 7%  32.4µs ± 2%
LogLineCopy/set=20-8   29.7µs ± 7%   29.6µs ± 2%   31.0µs ± 4%  31.8µs ± 5%  30.5µs ± 4%

name \ alloc/op       bench.len+50  bench.len+25  bench.len+10  bench.len    bench.nil
LogLineCopy/set=0-8    24.5kB ± 0%   24.5kB ± 0%   24.5kB ± 0%  24.5kB ± 0%  24.5kB ± 0%
LogLineCopy/set=1-8    31.3kB ± 0%   34.7kB ± 0%   34.4kB ± 0%  41.6kB ± 0%  39.0kB ± 0%
LogLineCopy/set=10-8   31.3kB ± 0%   34.7kB ± 0%   34.4kB ± 0%  41.6kB ± 0%  39.0kB ± 0%
LogLineCopy/set=20-8   31.3kB ± 0%   34.7kB ± 0%   34.4kB ± 0%  42.0kB ± 0%  39.3kB ± 0%

name \ allocs/op      bench.len+50  bench.len+25  bench.len+10  bench.len    bench.nil
LogLineCopy/set=0-8      3.00 ± 0%     3.00 ± 0%     3.00 ± 0%    3.00 ± 0%    3.00 ± 0%
LogLineCopy/set=1-8      4.00 ± 0%     5.00 ± 0%     5.00 ± 0%    7.00 ± 0%    6.00 ± 0%
LogLineCopy/set=10-8     4.00 ± 0%     5.00 ± 0%     5.00 ± 0%    7.00 ± 0%    6.00 ± 0%
LogLineCopy/set=20-8     4.00 ± 0%     5.00 ± 0%     5.00 ± 0%    7.00 ± 0%    6.00 ± 0%
```

Reply:
I still need to add the comment explaining why len+50%